tests/common/test_utils.py

```diff
@@ -42,6 +46,39 @@ def test_datetime_to_msec(self):
                 self.assertEqual(utils.datetime_to_msec(inval), expected)
 
 
+class FindTest(FileTestCase):
+    def setUp(self):
+        self.setUpPyfakefs()
+
+    def test_find(self):
+        test_1 = [
+            self.fs.create_file("/var/data.dat"),
+            self.fs.create_file("/var/data/xx1.txt"),
+            self.fs.create_file("/var/data/xx2.txt"),
+        ]
+
+        with mock.patch.object(VFS, "ls", return_value=test_1):
+            with mock.patch.object(VFS, "is_dir", return_value=True) as mock_is_dir:
+                mock_is_dir.side_effect = lambda f: self.fs.isdir(f.name)
```

Comment:
Though I did notice this while scrolling through: Instead of setting up a bunch of mocks and using a fake filesystem, it seems like it would be simpler to create a temporary directory:

```python
with tempfile.TemporaryDirectory() as tmp_name:
    tmp = pathlib.Path(tmp_name)
    (tmp / "data.dat").write_text("test")
    data_dir = tmp / "data"
    data_dir.mkdir()
    (data_dir / "xx1.txt").write_text("test")
    ...
```

The temporary directory will be cleaned up at the end of the with-block, and there is no need to muck with TileDB internals.

Reply:
I agree for `find` this is easier than mocking the file system.